js/private/js_library.bzl

```diff
@@ -0,0 +1,106 @@
+"""Expose some files with DeclarationInfo, like filegroup but can be a dep of ts_project
```

Comment:
I'm not quite convinced such a thing should be named `js_library` but we can merge and then discuss renaming again